### PR TITLE
Replace HttpClient with HuaweiCloud SDK in FlexibleEngine Plugin

### LINKED ISSUE #1168: replace http client from java-sdk-core in flexbileEngine plugin
Currently the flexible engine plugin uses the http client from java-sdk-core and this must be replaced and we must use our own http client. 

- [ ] create http client using spring rest client. 
- [ ] write the request signer logic. 
- [ ] remove all dependencies to java-sdk-core from SBOM.

### PR DESCRIPTION
**Pull request by baixinsui**
fixes #1168

### CODE REVIEW

File: plugins/flexibleengine/src/main/java/org/eclipse/xpanse/plugins/flexibleengine/common/FlexibleEngineClient.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            return false;
+        }
+        int statusCode = ((ServiceResponseException) ex).getHttpStatusCode();
+        return statusCode == ERROR_CODE_TOO_MANY_REQUESTS
+                || statusCode == ERROR_CODE_INTERNAL_SERVER_ERROR;
+    }
+
+    private HcClient getHcClient(ICredential credential, String servicePrefix, String regionName) {
+        String endpoint = PROTOCOL_HTTPS + servicePrefix + regionName + ENDPOINT_SUFFIX;
+        Region region = new Region(regionName, endpoint);
```
Comment (swaroopar) [inline plugins/flexibleengine/src/main/java/org/eclipse/xpanse/plugins/flexibleengine/common/FlexibleEngineClient.java:93]: this region variable is not used anywhere.

File: plugins/flexibleengine/src/main/java/org/eclipse/xpanse/plugins/flexibleengine/common/FlexibleEngineClient.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                    .withProjectId(projectId);
+        }
+        return credential;
+    }
+
+    private String getProjectId(ICredential credential, String regionName) {
+        String projectId = null;
+        try {
+            String endpoint = PROTOCOL_HTTPS + IAM_ENDPOINT_PREFIX + regionName + ENDPOINT_SUFFIX;
+            Region region = new Region(regionName, endpoint);
```
Comment (swaroopar) [inline plugins/flexibleengine/src/main/java/org/eclipse/xpanse/plugins/flexibleengine/common/FlexibleEngineClient.java:116]: same here. Not used. Please check other methods too in this class.

File: plugins/huaweicloud/pom.xml [hunk 3]
```diff
@@ -18,6 +18,19 @@
         <relativePath>../pom.xml</relativePath>
     </parent>
 
+    <repositories>
+        <repository>
+            <id>HuaweiCloudSDK</id>
+            <url>https://repo.huaweicloud.com/repository/maven/huaweicloudsdk/</url>
```
Comment (swaroopar) [inline plugins/huaweicloud/pom.xml:24]: as far i see, we do need this internal repository. All huaweicloud SDKs must be available in the maven central already. 
Please check this in the flexibleengine pom as well.

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: plugins/flexibleengine/src/main/java/org/eclipse/xpanse/plugins/flexibleengine/common/FlexibleEngineClient.java (hunk 2 above)
Comment (baixinsui) [reply plugins/flexibleengine/src/main/java/org/eclipse/xpanse/plugins/flexibleengine/common/FlexibleEngineClient.java]: Removed.

File: plugins/flexibleengine/src/main/java/org/eclipse/xpanse/plugins/flexibleengine/common/FlexibleEngineClient.java (hunk 1 above)
Comment (baixinsui) [reply plugins/flexibleengine/src/main/java/org/eclipse/xpanse/plugins/flexibleengine/common/FlexibleEngineClient.java]: Removed.

File: plugins/huaweicloud/pom.xml (hunk 3 above)
Comment (baixinsui) [reply plugins/huaweicloud/pom.xml]: Removed the internal repository.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**